cmd/manager/scap.go

```diff
@@ -223,6 +244,51 @@ func getPathFromWarningXML(in *xmlquery.Node, valueList map[string]string) []uti
 	return path
 }
 
+// Fetch all nodes from the cluster and find all roles for each node.
+func fetchNodesWithRole(ctx context.Context, c runtimeclient.Client) (map[string][]string, error) {
+	nodeList := v1.NodeList{}
+	if err := c.List(ctx, &nodeList); err != nil {
+		return nil, fmt.Errorf("failed to list nodes: %w", err)
+	}
+
+	roleNodesList := make(map[string][]string)
+	for _, node := range nodeList.Items {
+		nodeName := node.Name
+		nodeRoles := utils.GetFirstNodeRole(node.ObjectMeta.Labels)
+		roleNodesList[nodeRoles] = append(roleNodesList[nodeRoles], nodeName)
+	}
+
+	return roleNodesList, nil
+
+}
+
+// Get resourcePath for KubeletConfig
+func getKubeletConfigResourcePath(roleNodesList map[string][]string) []utils.ResourcePath {
+	resourcePath := []utils.ResourcePath{}
+	for role, nodeList := range roleNodesList {
+		for _, node := range nodeList {
+			resourcePath = append(resourcePath, utils.ResourcePath{
+				ObjPath:  "/api/v1/nodes/" + node + "/proxy/configz",
+				DumpPath: kubeletConfigPathPrefix + role + "/" + node,
+				Filter:   `.kubeletconfig|.kind="KubeletConfiguration"|.apiVersion="kubelet.config.k8s.io/v1beta1"`,
+			})
+		}
+	}
+	return resourcePath
+}
+
+// Get role name and node name from DumpPath
+func getRoleNodeNameFromDumpPath(dumpPath string) (string, string) {
+	if strings.HasPrefix(dumpPath, kubeletConfigPathPrefix) {
+		dumpPathSplit := strings.Split(dumpPath, "/")
+		if len(dumpPathSplit) != 4 {
+			return "", ""
+		}
+		return dumpPathSplit[2], dumpPathSplit[3]
```

Comment:
can you add an example of the dump path? Otherwise it's not easy to see what the magic constants are